ParquetRoller/Roller.cs

```diff
@@ -413,9 +413,31 @@ private static ExitCode ListMaxIDs(ModelCollection inWorkload)
         /// <returns><see cref="ExitCode.BadArguments"/></returns>
         private static ExitCode ListTags(ModelCollection inWorkload)
         {
-            // TODO This is a stub.
-            Console.WriteLine(inWorkload);
-            return ExitCode.BadArguments;
+            if (inWorkload == null || inWorkload.Count == 0)
+            {
+                Console.WriteLine("No defined content.");
+                return ExitCode.Success;
+            }
+
+            HashSet<ModelTag> allTags = new HashSet<ModelTag>();
+
+            foreach (var model in inWorkload)
+            {
+                foreach (var modelTag in model.GetAllTags())
+                {
+                    if (!allTags.Any(tag => tag.CompareTo(modelTag) == 0))
+                    {
+                        allTags.Add(modelTag);
+                    }
+                }
+            }
+
+            foreach (var modelTag in allTags.Distinct())
```

Comment:
Given that allTags is a HashSet and that you only add tags that do not compare as equivalent, shouldn't these already be distinct?

Reply:
Oh, true! Removed.